improver/utilities/spatial.py

```diff
@@ -166,9 +168,44 @@ class DifferenceBetweenAdjacentGridSquares(BasePlugin):
     """
 
     @staticmethod
-    def _axis_wraps_around_meridian(axis: Coord, cube: Cube):
+    def _axis_wraps_around_meridian(axis: Coord, cube: Cube) -> bool:
+        """Returns true if the cube is 'circular' with the given axis wrapping around, i.e. if there
+        is a smooth transition between 180 degrees and -180 degrees on the axis.
+
+        Args:
+            axis:
+                Axis to check for circularity.
+            cube:
+                The cube to which the axis belongs.
+
+        Returns:
+            True if the axis wraps around the meridian; false otherwise.
+        """
         return axis.circular and axis == cube.coord(axis="x")
 
+    @staticmethod
+    def _get_wrap_around_mean_point(points: ndarray) -> float:
+        """
+        Calculates the midpoint between the two x coordinate points nearest the meridian.
+
+        args:
+            points:
+                The x coordinate points of the cube.
+
+        returns:
+            The x value of the midpoint between the two x coordinate points nearest the meridian.
+        """
+        if np.min(points) < 0:
+            min_azimuth = -180
+            max_azimuth = 180
+        else:
+            min_azimuth = 0
+            max_azimuth - 360
```

Comment:
I've gone back and forth on this so I'm happy to be over-ruled by the second reviewer but I'm going to say that  we only worry about handling the -180 to 180 case. However, it would still be worthwhile making sure that the plugin handles other circular grids by ignoring the wrap around point and this will need to be tested